File: 523-continuous-subarray-sum/523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        
        #prefix sum
        for i in range(1,len(nums)):
            nums[i]=nums[i]+nums[i-1]

        #finding remainder
        remainder=[i%k for i in nums]

        #value with sum zero has index -1
        d={0:-1}

        #storing first frequency of remainder in dictionary
        for i in range(len(remainder)):
            if  remainder[i] not in d:
                d[remainder[i]]=i
        
        
        for i in range(len(remainder)):
            if remainder[i] in d:
                if i - d[remainder[i]] > 1:
                    return True
        return False

File: 523-continuous-subarray-sum/523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        HashMap<Integer,Integer> d=new HashMap<>();
        
        // prefix sum array
        for(int i=1;i<nums.length;i++)
            nums[i]=nums[i]+nums[i-1];
        
        // array having remainder of prefix sum
        for(int i=0; i<nums.length;i++)
            nums[i]=nums[i]%k;
        
        d.put(0,-1);
        // storing first occurence of remainder in dict.
        for (int i=0;i<nums.length;i++)
        {
            if(d.containsKey(nums[i]))
                continue;
            else
                d.put(nums[i],i);
        }
        for (int i=0;i<nums.length;i++)
        {
            if (d.containsKey(nums[i]))
                if (i-d.get(nums[i])>1)
                    return true;
        }
        return false;
    }
}